[FW][MIG] l10n_uy_edi_stock: version 17.0

fw-bot-adhoc opened the pull request:
- cosas que movimos de l10n_uy_edi que nos pidieron quitaramos
- adaptando modelo cfe, y trayendo tolo que no nos aceptaron adaptandolo para que funcione como un modelo normal y no un mixin
- get dgi state button
- update methods and field renamed
- move stuff from l10n uy edi official
- UY transport
- add constraint not sure yet
- place of delivery boolean
- TipoTraslado
- move template to views
- replace fields

---
Task 37187 (stock)

Forward-Port-Of: ingadhoc/uruguay#244

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/uruguay/pull/311).

fw-bot-adhoc commented:
@zaoral @pablohmontenegro cherrypicking of pull request ingadhoc/uruguay#244 failed.

* 8e111ab68cdf3ff8a28f6b2e4747678db571d610 <- on this commit
* 89bb8991238f0d38b48c5f01e1ef8e33a4b02645

stdout:
```
Auto-merging l10n_uy_edi_stock/models/stock_picking.py
CONFLICT (content): Merge conflict in l10n_uy_edi_stock/models/stock_picking.py
Auto-merging l10n_uy_edi_stock/__manifest__.py
CONFLICT (content): Merge conflict in l10n_uy_edi_stock/__manifest__.py

```

stderr:
```
18:46:13.878319 git.c:444               trace: built-in: git cherry-pick 8e111ab68cdf3ff8a28f6b2e4747678db571d610
error: could not apply 8e111ab... [MIG] l10n_uy_edi_stock: version 17.0
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'
----------
status:

```

Either perform the forward-port manually (and push to this branch, proceeding as usual) or close this PR (maybe?).

In the former case, you may want to edit this PR message as well.

More info at https://github.com/odoo/odoo/wiki/Mergebot#forward-port

zaoral committed:
[MIG] l10n_uy_edi_stock: version 18.0

Make some adaptations to make it work with v18.0 and also
fix with pre-commit

original commit 8e111ab68cdf3ff8a28f6b2e4747678db571d610

zaoral committed:
[ADD] l10n_uy_ux: make it compatible with e-remitos

- change related not store field to compute not stored fields: the
  ones realeted to the origin of the cfe (could be move/picking record)
  The fields changed where: l10n_latam_document_type_id, l10n_latam_document_number,
  company_id and partner_id
- hide the move field is this one is empty, this way we can use
  the view for both moves and pickings
- now that we have a lot of docs types it is better to show the doc
  prefix in the CFE name to identify them better.
- remove required True in apply_on, not necessary and giving us problem
  with the edi stock module
- add a workaround to make xml and pdf naming on the move possible
  because with the new change of the compute fields was not working.

jue-adhoc reviewed (approved):
@zaoral kathy! lo veo bien al PR. Espero el OK de Juli por las dudas para ver que esté funcionando todo ok y mezclamos!

jue-adhoc commented:
@roboadhoc r+
